pypdf/filters.py

```diff
@@ -444,11 +441,11 @@ def decode(self) -> str:
                 else:
                     if cW < self.dictlen:
                         baos += self.dict[cW]
-                        p = self.dict[pW] + self.dict[cW][0]
+                        p = self.dict[pW] + self.dict[cW][0:1]
```

Comment:
Why do we need this non-functional change?

Reply:
as self.dict[] contains bytes, self.dict[cW][0] returns int and self.dict[cW][0:1] returns bytes (of len 1). int is invalid for the concatenation